addons/utm/models/utm_mixin.py

```diff
@@ -62,7 +62,7 @@ def _find_or_create_record(self, model_name, name):
         """Based on the model name and on the name of the record, retrieve the corresponding record or create it."""
         Model = self.env[model_name]
 
-        record = Model.with_context(active_test=False).search([('name', '=', name)], limit=1)
+        record = Model.with_context(active_test=False).search([('name', '=ilike', name)], limit=1)
```

Comment:
problem with this is that if I create both `FaceBook` and `Facebook`, it will just pick whatever happens to be first in DB instead of the correct campaign.

Ideally, we wouldn't change this to ilike at all, so older records can still be uniquely identified.
~~But if not, we should remove the limit on the search and order by exact equality, then pick the first. So that if there's one exactly equal, we pick that one, otherwise we pick a similar one.~~ <- that's one fix to this particular issue but I'm not a fan and it doesn't do anything for backend

I think a good way to mitigate this is to ensure we have unique names case-insensitively. i.e. either enforce lowercase or otherwise make `_get_unique_names` add `[2]` even if the records don't have the name name case-sensitively. So either of:
- `facebook`, `facebook [1]`
- `FaceBook`, `Facebook [1]`

Reply:
@reth-odoo what do u think of this?